### PR TITLE
Fixes a few runtimes

### PR DESCRIPTION
**Pull request by ike709**
Runtimes bad.

## Changelog
:cl:
fix: Null clients are now less likely to completely break roundend.
fix: Fixed some internal affairs objective code just... not working.
fix: Fixed two ambience runtimes.
fix: Radios will no longer runtime when we can't get their turf.
/:cl:

### CODE REVIEW

File: code/__HELPERS/roundend.dm [hunk 1]
```diff
@@ -200,7 +200,7 @@
 				var/mob/M = C.mob
 				if(M?.mind?.current && LAZYLEN(M.mind.crew_objectives))
 					for(var/datum/objective/crew/CO in M.mind.crew_objectives)
-						if(CO.check_completion())
+						if(C && CO.check_completion()) //Yes, the client can be null here. BYOND moment.
 							C.inc_metabalance(METACOIN_CO_REWARD, reason="Completed your crew objective!")
```
Comment (park66665) [inline code/__HELPERS/roundend.dm]:
```suggestion
						if(!C) //Yes, the client can be null here. BYOND moment.
							break
						if(CO.check_completion())
							C?.inc_metabalance(METACOIN_CO_REWARD, reason="Completed your crew objective!")
```

Comment (ike709) [reply code/__HELPERS/roundend.dm]: Shouldn't that be a break?

Comment (park66665) [reply code/__HELPERS/roundend.dm]: you commented after I edited it; refresh the browser

File: code/__HELPERS/roundend.dm [hunk 2]
```diff
@@ -200,8 +200,10 @@
 				var/mob/M = C.mob
```
Comment (ike709) [inline code/__HELPERS/roundend.dm]:
```suggestion
				var/mob/M = C ? C.mob : null
```

File: code/__HELPERS/roundend.dm [hunk 3]
```diff
@@ -197,11 +197,13 @@
 			C.process_endround_metacoin()
 
 			if(CONFIG_GET(flag/allow_crew_objectives))
-				var/mob/M = C.mob
+				var/mob/M = C ? C.mob : null
```
Comment (ike709) [inline code/__HELPERS/roundend.dm]:
```suggestion
				var/mob/M = C.mob
```

File: code/controllers/subsystem/ambience.dm [hunk 4]
```diff
@@ -70,6 +70,6 @@ SUBSYSTEM_DEF(ambience)
 /datum/controller/subsystem/ambience/proc/update_effects(mob/M) // Effect, random sounds that will play at random times, IC (requires the user to be able to hear)
 	var/area/A = get_area(M)
 
-	if (A.ambient_effects && (M.client.prefs.toggles & SOUND_AMBIENCE) && M.can_hear_ambience() && (world.time - M.client.ambient_effect_last_played) > AMBIENT_EFFECT_COOLDOWN && prob(5) && !M.client.channel_in_use(CHANNEL_AMBIENT_EFFECTS)) // 1/20 chance to play every second after cooldown
+	if (length(A.ambient_effects) && (M.client.prefs.toggles & SOUND_AMBIENCE) && M.can_hear_ambience() && (world.time - M.client.ambient_effect_last_played) > AMBIENT_EFFECT_COOLDOWN && prob(5) && !M.client.channel_in_use(CHANNEL_AMBIENT_EFFECTS)) // 1/20 chance to play every second after cooldown
+		M?.client?.ambient_effect_last_played = world.time
```
Comment (park66665) [inline code/controllers/subsystem/ambience.dm]:
```suggestion
	if (length(A.ambient_effects) && (M.client?.prefs.toggles & SOUND_AMBIENCE) && M.can_hear_ambience() && (world.time - M.client?.ambient_effect_last_played) > AMBIENT_EFFECT_COOLDOWN && prob(5) && !M.client?.channel_in_use(CHANNEL_AMBIENT_EFFECTS)) // 1/20 chance to play every second after cooldown
		M.client?.ambient_effect_last_played = world.time
```

`M` is also checked before calling this proc, but unlike `M.client`, it is much more likely to persist.

File: code/game/objects/items/devices/radio/radio.dm [hunk 5]
```diff
@@ -258,7 +258,7 @@
 	var/turf/position = get_turf(src)
 	for(var/obj/item/jammer/jammer in GLOB.active_jammers)
 		var/turf/jammer_turf = get_turf(jammer)
-		if(position.z == jammer_turf.z && (get_dist(position, jammer_turf) <= jammer.range))
+		if(position && position.z == jammer_turf.z && (get_dist(position, jammer_turf) <= jammer.range))
```
Comment (park66665) [inline code/game/objects/items/devices/radio/radio.dm]:
```suggestion
		if(position?.z == jammer_turf.z && (get_dist(position, jammer_turf) <= jammer.range))
```

if `position` is null then `position?.z` will return null.

**Review by park66665 — APPROVED** [review]
Things look a little bit band-aid-ish, but that is fine.

**Pull request merged**